### PR TITLE
Use DescribeInstance to retrieve filtered AWS instances

### PR DESCRIPTION
**Pull request by trjstewart**
### Proposed changes
This is an initial implementation to satisfy the minimum requirements of issue #28. It would also close out #6.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-asg-sync/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Comment by trjstewart** [comment]
no tests were added as this initial implementation doesn't change the interface for the config file.
also closes PR #6.
refer to issue #29 for further comments.

File: cmd/sync/aws.go [hunk 1]
```diff
@@ -111,79 +106,58 @@ func parseAWSConfig(data []byte) (*awsConfig, error) {
...
-		return nil, err
+		return false, fmt.Errorf("couldn't check if an AutoScaling group exists: %v", err)
 	}
 
-	var result []string
-	for _, ins := range instances {
-		if len(ins.NetworkInterfaces) > 0 && ins.NetworkInterfaces[0].PrivateIpAddress != nil {
-			result = append(result, *ins.NetworkInterfaces[0].PrivateIpAddress)
-		}
+	if len(response.Reservations) == 0 {
```
Comment (pleshakov) [inline cmd/sync/aws.go]: this could be simplified:
```go
return len(response.Reservations) > 0, nil
```

File: examples/aws.md [hunk 2]
```diff
@@ -34,6 +38,6 @@ upstreams:
 * The `region` key defines the AWS region where we deploy NGINX Plus and the Auto Scaling groups. Setting `region` to `self` will use the EC2 Metadata service to retreive the region of the current instance.
 * The `upstreams` key defines the list of upstream groups. For each upstream group we specify:
   * `name` – The name we specified for the upstream block in the NGINX Plus configuration.
-  * `autoscaling_group` – The name of the corresponding Auto Scaling group.
+  * `autoscaling_group` – The name of the corresponding Auto Scaling group. Use of wildcards is supported.
```
Comment (pleshakov) [inline examples/aws.md]:
```suggestion
  * `autoscaling_group` – The name of the corresponding Auto Scaling group. Use of wildcards is supported. For example, `backend-*`.
```

File: examples/aws.md [hunk 3]
```diff
@@ -26,6 +26,10 @@ upstreams:
    autoscaling_group: backend-two-group
    port: 80
    kind: http
+ - name: backend
```
Comment (pleshakov) [inline examples/aws.md]: I suggest removing this snippet. The example config reflects the setup described in the main readme -- https://github.com/nginxinc/nginx-asg-sync#nginx-plus-configuration , which doesn't include the third upstream,

**Comment by trjstewart** [comment]
Thanks for the suggestions @pleshakov. I've fixed those few things up.

**Comment by pleshakov** [comment]
@trjstewart thanks! the changes look good. we'll finish the testing shortly and merge it.

@Rulox yep, we'll include this feature into a new release. 

For the new release, we're also planning to include support for upstream servers parameter (max_fails, max_conns, fail_timeout, slow_start), which we're currently working on.

**Comment by trjstewart** [comment]
@pleshakov do you have an ETA on getting the new release out? Just trying to decide if I want to build the binary myself and put it on our CDN or wait.

**Comment by pleshakov** [comment]
@trjstewart the ETA is 2 weeks.

**Comment by trjstewart** [comment]
@pleshakov I'd be happy to take a look at adding the server parm support if you think that would speed things up and make less work for you guys?

**Comment by Rulox** [comment]
@trjstewart thanks for your help! We really appreciate it.

I am not sure if the release date can be earlier regarding the time the new features are available (we have the team working on these features already). @pleshakov maybe you have more insights on this?

In the meantime, this PR is good to go, so merging it now! :shipit:

**Comment by pleshakov** [comment]
@trjstewart as Raul mentioned, we're already working on the feature. thanks for your help.

**Comment by Rulox** [comment]
@trjstewart If you are still interested in this, this feature (upstream parameters) has been rolled out on the 0.4-1 version, sorry for the delay.

**Pull request merged**